src/core/EventManager.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { MembershipManager } from "./MembershipManager.sol";
+
+contract EventManager is Ownable {
+    uint256 private eventId;
+    MembershipManager public memberShipManager;
+    using SafeERC20 for ERC20;
+
+    /**
+     * @notice Maps eventId to list of registered users
+     */
+    mapping(uint256=>address[]) public registeredUsers;
```

Comment:
I wonder if making it a mapping of a mapping will be a much cheaper operation than operating on arrays:

```
mapping (uint256 => mapping(address => bool) public isRegistered;
```

Not sure if it will save that much but maybe is saves some gas when registered users increase. Though we are not enumerating over the array in the code anyways.